homeassistant/components/media_player/squeezebox.py

```diff
@@ -0,0 +1,277 @@
+
+Configuration:
+
+To use SqueezeBox add something like this to your configuration:
+
+media_player:
+  platform: squeezebox
+  name: SqueezeBox
+  server: 192.168.1.21
+  player: Player1
+  port: 9090
+  user: user
+  password: password
+
+Variables:
+
+name
+*Optional
+The name of the device
+
+server
+*Required
+The address of the Logitech Media Server
+
+player
+*Required
+The unique name of the player
+
+port
+*Optional
+Telnet port to Logitech Media Server, default 9090
+
+user
+*Optional
+User, if password protection is enabled
+
+password
+*Optional
+Password, if password protection is enabled
+"""
+
+import logging
+import telnetlib
+import urllib.parse
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, SUPPORT_PAUSE, SUPPORT_SEEK, SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_MUTE, SUPPORT_PREVIOUS_TRACK, SUPPORT_NEXT_TRACK,
+    MEDIA_TYPE_MUSIC)
+from homeassistant.const import (
+    STATE_IDLE, STATE_PLAYING, STATE_PAUSED, STATE_OFF, STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_SQUEEZEBOX = SUPPORT_PAUSE | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE |\
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | SUPPORT_SEEK
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the squeezebox platform. """
+    add_devices([
+        SqueezeBoxDevice(
+            config.get('name', 'SqueezeBox'),
+            config.get('server'),
+            config.get('player'),
+            config.get('port', '9090'),
+            config.get('user', None),
+            config.get('password', None)
+        )])
+
+
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=too-many-public-methods
+class SqueezeBoxDevice(MediaPlayerDevice):
+    """ Represents a SqueezeBox device. """
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, name, server, player, port, user, password):
+        super(SqueezeBoxDevice, self).__init__()
+        self._name = name
+        self._server = server
+        self._player = player
+        self._port = port
+        self._user = user
+        self._password = password
+        self._status = {}
+        self.update()
+        self._http_port = self._query('pref', 'httpport', '?')
+
+    @property
+    def name(self):
+        """ Returns the name of the device. """
+        return self._name
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        if 'power' in self._status and self._status['power'] == '0':
+            return STATE_OFF
+        if 'mode' in self._status:
+            if self._status['mode'] == 'pause':
+                return STATE_PAUSED
+            if self._status['mode'] == 'play':
+                return STATE_PLAYING
+            if self._status['mode'] == 'stop':
+                return STATE_IDLE
+        return STATE_UNKNOWN
+
+    def update(self):
+        self._get_status()
```

Comment:
Why not move the logic from _get_status to update and remove the _get_status method? Is there a reason that the _get_status method must exist?